### PR TITLE
Sanity v3 upgrade

### LINKED ISSUE #159: Upgrade to Sanity Studio V3
Upgrade to Sanity Studio V3, which can be embedded as part of the Next.js site and served off a route of the Next.js site. This should also allow us to un-monorepo our site, since we'll now just have "one site".

### PR DESCRIPTION
**Pull request by umxr**
ive upgraded sanity to V3, in v3 the main change are the following:

- Creating Documents/Objects
- defining config

**Creating Documents/Objects**
in V3, we create documents using the `defineType` function from sanity. this function takes in the same object structure you would normally use when creating documents/objects. Creating fields comes with a slight change aswell. we use the `defineField` function. Similar to creating v2 documents, the function takes in an object in the same shape as we used to create v2 fields

**Defining Config**
in v2 we had a sanity.json file which allowed us to customise our studio instance. this has been replaced in v3, with a `sanity.config.ts` file. to create a config export a default function calling `defineConfig`

closes #159

### CODE REVIEW

File: packages/sanity/schemas/product.ts
Comment (gksander) [inline packages/sanity/schemas/product.ts:51]: Apologies for the timing conflict here, but @sarmeyer recently merged [this PR](https://github.com/FormidableLabs/nextjs-sanity-fe/pull/164) that made some changes to the way images are handled to be a bit more idiomatic. Can we cherry-pick those changes into this PR so we don't accidentally undo her work?

**Review by gksander — CHANGES_REQUESTED** [review]
This looks amazing – thanks for taking this on! Before merging this, I just want to make sure we incorporate some work Sarah had done while you were working on this (apologies on my part for the timing there).

File: packages/sanity/utils/sanityClient.ts
Comment (umxr) [reply packages/sanity/utils/sanityClient.ts:11]: Understood, ill make this change. thanks!

File: packages/sanity/schemas/product.ts
Comment (umxr) [reply packages/sanity/schemas/product.ts:51]: Sure thing! ill take a look at her PR and amend my pr :)

Comment (umxr) [reply packages/sanity/schemas/product.ts]: @sarmeyer ive updated this branch with the changes you added in #164, can you please review them to make sure ive not missed anything. 

thanks in advance :)

File: packages/sanity/package.json
Comment (sarmeyer) [inline packages/sanity/package.json:26]: nice to get this bump as well!

**Comment by sarmeyer** [comment]
There is one breaking change we need to account for in the `sanity/package.json` - we'll need to switch the `dev` script to `sanity dev`

image

File: packages/sanity/schemas/variant.ts
Comment (sarmeyer) [inline packages/sanity/schemas/variant.ts:139]: due to the `productImage` changes, this should be accessing the correct image on `asset` rather than `images`
```suggestion
      media: "images.0.asset",
```

**Review by gksander — APPROVED** [review]
Whoops, sorry for holding this up! Looks good!

File: packages/sanity/schemas/flavour.ts
Comment (luisabarca) [inline packages/sanity/schemas/flavour.ts:23]: for SEO purpose, slug should be 75-120 max

**Review by luisabarca — APPROVED** [review]
👍

**Pull request merged**
